src/hooks/useSortedRoutesWithQuotes.ts

```diff
@@ -102,48 +102,44 @@ export const useSortedRoutesWithQuotes = ({
 
   // Only routes with quotes are sorted.
   const sortedRoutesWithQuotes = useMemo(() => {
-    return bucketByEta(routesWithQuotes)
-      .map((bucket) => {
-        return bucket.sort((routeA, routeB) => {
-          const routeConfigA = config.routes.get(routeA.route);
-          const routeConfigB = config.routes.get(routeB.route);
+    return routesWithQuotes.sort((routeA, routeB) => {
+      const routeConfigA = config.routes.get(routeA.route);
+      const routeConfigB = config.routes.get(routeB.route);
+
+      // Prioritize preferred route to avoid flickering the UI
+      // when the preferred route gets autoselected
+      if (preferredRouteName) {
+        if (routeA.route === preferredRouteName) {
+          return -1;
+        } else if (routeB.route === preferredRouteName) {
+          return 1;
+        }
+      }
 
-          // Prioritize preferred route to avoid flickering the UI
-          // when the preferred route gets autoselected
-          if (preferredRouteName) {
-            if (routeA.route === preferredRouteName) {
-              return -1;
-            } else if (routeB.route === preferredRouteName) {
-              return 1;
-            }
-          }
+      // 1. Sort by ETA (fastest first)
+      const etaA = routeA.quote.eta ?? Infinity;
```

Comment:
If ETA is not there, it can also mean instant. For example same chain swaps is considered instant.